Source/Graph/Graph.cs

```diff
@@ -1182,13 +1182,12 @@ public static IEnumerable<object> FindReachableNodesInGraphWithMergeNodes(Dictio
           continue;
         }
         
-        if (node is IReadOnlySet<object> mergeNode) {
-          if (!visitedEdges.IsSupersetOf(mergeNode)) {
+        if (node is IEnumerable<object> objects) {
+          if (!visitedEdges.IsSupersetOf(objects)) {
             continue;
           }
-        } else {
-          visitedEdges.Add(node);
         }
+        visitedEdges.Add(node);
```

Comment:
Accurate catching of a bug, where merge nodes would not be added if they were visitable.
Since the check is now against IEnumerable instead of IReadOnlySet, please update the comment accordingly.